cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache.go

```diff
@@ -0,0 +1,324 @@
+/*
+Copyright © 2021 VMware
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	corev1 "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1"
+	"github.com/kubeapps/kubeapps/cmd/kubeapps-apis/server"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	log "k8s.io/klog/v2"
+)
+
+type fluxPlugInCache struct {
+	// to prevent multiple watchers
+	watcherStarted bool
+	// this mutex guards watcherStarted var
+	watcherMutex sync.Mutex
+	redisCli     *redis.Client
+	clientGetter server.KubernetesClientGetter
+	// these waitGroup is used exclusively by unit tests to block until all expected repos have
+	// been indexed by the go routine running in the background. The creation of the WaitGroup object
+	// and .Add() is expected to be done by the unit test client. The server-side only signals
+	// when indexing one repo is complete
+	indexRepoWaitGroup *sync.WaitGroup
+}
+
+func newCache(clientGetter server.KubernetesClientGetter) (*fluxPlugInCache, error) {
+	log.Infof("+newCache")
+	REDIS_ADDR, ok := os.LookupEnv("REDIS_ADDR")
+	if !ok {
+		return nil, status.Errorf(codes.FailedPrecondition, "missing environment variable REDIS_ADDR")
+	}
+	REDIS_PASSWORD, ok := os.LookupEnv("REDIS_PASSWORD")
+	if !ok {
+		return nil, status.Errorf(codes.FailedPrecondition, "missing environment variable REDIS_PASSWORD")
+	}
+	REDIS_DB, ok := os.LookupEnv("REDIS_DB")
+	if !ok {
+		return nil, status.Errorf(codes.FailedPrecondition, "missing environment variable REDIS_DB")
+	}
+
+	REDIS_DB_NUM, err := strconv.Atoi(REDIS_DB)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("newCache: addr: [%s], password: [%s], DB=[%d]", REDIS_ADDR, REDIS_PASSWORD, REDIS_DB)
+
+	return newCacheWithRedisClient(
+		clientGetter,
+		redis.NewClient(&redis.Options{
+			Addr:     REDIS_ADDR,
+			Password: REDIS_PASSWORD,
+			DB:       REDIS_DB_NUM,
```

Comment:
yes, I was wondering the same exact thing. Instead of a 'namespace' which redis doesn't really have a concept of, use a different DB. Will need to see what the cost is, if any, of having multiple concurrent DBs